sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -688,6 +691,15 @@ public void validate() {
     private Instant curTimestamp;
     private Iterator<PartitionState> curBatch = Collections.emptyIterator();
 
+    private final Counter messagesConsumed = Metrics.counter("io", "messagesConsumed");
+    private final Counter bytesConsumed = Metrics.counter("io", "bytesConsumed");
+    private final Distribution backlogBytes = Metrics.distribution("io", "backlog.bytes");
+    private final Distribution backlogMessages = Metrics.distribution("io", "backlog.messages");
```

Comment:
Aviem -- it sounds like adding something more like a "push" Gauge (allowing us to push values into it from the code) may be appropriate. Would it make sense to create a separate ticket and implement that first?